### ISSUE TITLE
error <SNR>12_VimDebuggerLoad and <SNR>12_VimDebuggerInit

### ISSUE THREAD

**Issue by jesusangelm** [spf13/spf13-vim#30]
error running vim command after installing spf13-vim
using Ubuntu 10.10 64bits, Vim 7.3

$  vim
Se ha detectado un error al procesar function <SNR>12_VimDebuggerLoad:
línea    2
E117: Función desconocida: <SNR>12_VimDebuggerInit
Se ha detectado un error al procesar function DefPython:
línea  550
Traceback (most recent call last):
  File "<string>", line 547, in <module>
  File "<string>", line 397, in **init**
Vim: Capté una señal mortal SEGVger' is not defined

Vim: Finalizado.
[1]    18945 segmentation fault  vim

Opening Gvim i see this error:

$  gvim
Se ha detectado un error al procesar /home/angel/.vimrc:
línea  429
E518: Opción desconocida: transparency=5
Se ha detectado un error al procesar function <SNR>14_VimDebuggerLoad:
línea    2
E117: Función desconocida: <SNR>14_VimDebuggerInit
Se ha detectado un error al procesar function DefPython:
línea  550
Traceback (most recent call last):
  File "<string>", line 547, in <module>
  File "<string>", line 397, in **init**
Vim: Capté una señal mortal SEGVger' is not defined

Vim: Finalizado.
[1]    23219 segmentation fault  gvim

**Comment by wellran** [comment]
I got the same errors.
I use Ubuntu 11.04(32 bits),vim 7.3

**Comment by chtaube** [comment]
Seems I have the same problem with Vim 7.2.445 on Debian 6.0.2:

```
$ LC_ALL=C vim
Error detected while processing function <SNR>12_VimDebuggerLoad:
line    2:
E117: Unknown function: <SNR>12_VimDebuggerInit
Error detected while processing function DefPython:
line  550:
Traceback (most recent call last):
  File "<string>", line 547, in <module>
  File "<string>", line 397, in __init__
NameError: global name 'VimDebugger' is not defined
Press ENTER or type command to continue
```

**Comment by markdwhite** [comment]
+1

Error detected while processing function <SNR>13_VimDebuggerLoad:
line    2:
E117: Unknown function: <SNR>13_VimDebuggerInit
Error detected while processing function DefPython:
line  550:
Traceback (most recent call last):
  File "<string>", line 547, in <module>
  File "<string>", line 397, in **init**
NameError: global name 'VimDebugger' is not defined

**Comment by vertex** [comment]
I'm having this issue too  Ubuntu 11.04 - the Natty Narwhal

**Comment by twmobius** [comment]
Check this for a possible solution. https://github.com/taxilian/VimDebugger/issues/12#issuecomment-1593173

**Comment by vuquochuy** [comment]
I also have same problem in Ubuntu 11.04, Vim 7.3

**Comment by ghost** [comment]
same issue with Arch Linux

Error detected while processing function <SNR>14_VimDebuggerLoad:
line    2:  
E117: Unknown function: <SNR>14_VimDebuggerInit  
Error detected while processing function DefPython:  
line  550:  
Traceback (most recent call last):  
  File "<string>", line 547, in <module>  
  File "<string>", line 397, in **init**  
NameError: global name 'VimDebugger' is not defined  
Taglist: Exuberant ctags (http://ctags.sf.net) not found in PATH. Plugin is not loaded.  
Press ENTER or type command to continue

**Comment by adgaudio** [comment]
The problem is that the function, VimDebuggerLoad, tries to call VimDebuggerInit before VimDebuggerInit is defined.  The solution is to call VimDebuggerLoad after VimDebuggerInit is defined.

I appended a patch to this post.  to apply it, save it to a file and then execute:

```
git apply --check FILENAME.patch # check patch won't cause conflicts
git am --signoff < FILENAME.patch # apply patch
```

The patch in question:

```
From 0a9b2fa821450ae423e87f8cfc9fe25b0d8d8f4d Mon Sep 17 00:00:00 2001
From: Alex Gaudio <adgaudio@gmail.com>
Date: Sat, 10 Sep 2011 00:11:30 -0400
Subject: [PATCH] bug fix: VimDebuggerLoad no longer calls VimDebuggerInit
 before it is defined

---
 plugin/VimDebugger.vim |   16 ++++++++--------
 1 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/plugin/VimDebugger.vim b/plugin/VimDebugger.vim
index f0c38cf..d5109b4 100644
--- a/plugin/VimDebugger.vim
+++ b/plugin/VimDebugger.vim
@@ -32,14 +32,6 @@
 "    Steve Francia <spf13-vim@spf13.com>


-function s:VimDebuggerLoad()
-  if has('python')
-    call s:VimDebuggerInit()
-  endif
-endfunction
-
-call s:VimDebuggerLoad()
-
 function s:VimDebuggerInit()

 if has('python')
@@ -75,6 +67,14 @@ if has('python')
 endif
 endfunction

+function s:VimDebuggerLoad()
+  if has('python')
+    call s:VimDebuggerInit()
+  endif
+endfunction
+
+call s:VimDebuggerLoad()
+
 function! g:__dbg_WatchFoldText()
   let nucolwidth = &fdc + &number*&numberwidth
   let winwd = winwidth(0) - nucolwidth - 5
-- 
1.7.6.1
```

**Comment by spf13** [comment]
Fixed in the 3.0 release of spf13-vim
